api/src/services/survey-service.ts

```diff
@@ -750,7 +750,30 @@ export class SurveyService extends DBService {
     await Promise.all(promises);
   }
 
-  async updateSurveyLocation(data: PutSurveyLocationData): Promise<void> {
+  async insertUpdateDeleteSurveyLocation(surveyId: number, data: PostSurveyLocationData[]) {
+    const existingLocations = await this.getSurveyLocationsData(surveyId);
+    // compare existing locations with passed in locations
+    // any locations not found in both arrays will be deleted
+    const deletes = existingLocations.filter(
+      (existing) => !data.find((incoming) => incoming?.survey_location_id === existing.survey_location_id)
+    );
+    const deletePromises = deletes.map((item) => this.deleteSurveyLocation(item.survey_location_id));
+
+    const inserts = data.filter((item) => !item.survey_location_id);
+    const insertPromises = inserts.map((item) => this.insertSurveyLocations(surveyId, item));
+
+    const updates = data.filter((item) => item.survey_location_id);
+    const updatePromises = updates.map((item) => this.updateSurveyLocation(item));
+
+    return [...insertPromises, ...updatePromises, ...deletePromises];
```

Comment:
I'd argue that wrapping this in `Promise.all` would make a more valuable return type

Reply:
good call